index.js

```diff
@@ -454,11 +454,14 @@ module.exports = class Corestore extends ReadyResource {
     })
 
     if (!this.passive) {
+      const muxer = stream.noiseStream.userData
+      muxer.cork()
```

Comment:
There are some potential edge cases here, where `uncork()` never gets called. Unsure if we should worry about them though, so mostly sanity checking.

The edge cases:
- `stream.noiseStream.opened` never resolves (I don't think that can't currently happen though)
- An error in this flow, after `muxer.cork()` and before `stream.noiseStream.opened.then(...)` runs. This might need a `try-finally` of some kind if errors there don't cause a crash, because an error would leave protomux in a corked state forever

Reply:
opened always resolves and the others are not allowed to throw, but let me dbl check

Reply:
yes its all good